NRLMMD-GEOIPS/geoips
# Create "output_checker_kwargs" Argument

# Reviewer Checklist

* [x] Required ***existing tests*** pass (full_test.sh)
* [x] NO Required ***unit tests*** (not implementing unit tests yet...)
* [x] Required ***integration tests*** added and pass for new/modified functionality (added to abi.static.Infrared script)
* [x] Required ***documentation*** added for new/modified functionality (commented)

# Related Issues
fixes #366 
fixes #293 
Branched off of #351 

# Testing Instructions
Run geoips/tests/integration_tests/full_test.sh and get an output 0 to ensure this is correct. It was on my side.

# Summary

With the addition of output_checkers, we decided we would also like to have flexibility
with our pixel comparisons, as well as any other types of comparisons. Therefore, we
need to add a commandline argument called "output_checker_kwargs" to handle this.
Specifically for images, we'd like to have the ability to adjust the "image_threshold"
(% difference between pixels), for different user needs. This update adds three
threshold values ["lenient" (10%), "medium" (5%), "strict" (0%)].

    - modified: geoips/commandline/args.py
    - modified: geoips/interfaces/module_based/output_checkers.py
    - modified: geoips/plugins/modules/output_checkers/image.py
    - modified: geoips/plugins/modules/output_checkers/geotiff.py
    - modified: geoips/plugins/modules/output_checkers/netcdf.py
    - modified: geoips/plugins/modules/output_checkers/text.py
    - modified: geoips/plugins/modules/procflows/config_based.py
    - modified: geoips/plugins/modules/single_source.py
